dialogue-clients/src/main/java/com/palantir/dialogue/clients/DnsSupport.java

```diff
@@ -106,6 +111,18 @@ static <I> Refreshable<DnsResolutionResults<I>> pollForChanges(
         return dnsResolutionResult;
     }
 
+    @Unsafe
+    @Nullable
+    static String tryGetHost(@Unsafe String uriString) {
+        try {
+            URI uri = new URI(uriString);
+            return uri.getHost();
+        } catch (URISyntaxException | RuntimeException e) {
+            log.debug("Failed to parse URI", e);
+            return null;
+        }
+    }
```

Comment:
Extracted from a lambda in `DialogueDnsResolutionWorker` to reuse within `describeHostname`. There's a similar method in `ReloadingClientFactory` which doesn't share this implementation because it's where we provide higher priority logging and metrics for invalid URIs -- we don't want to duplicate that N times for a single failure.